api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -1265,6 +1269,10 @@ public static MapType ofRequired(int keyId, int valueId, Type keyType, Type valu
     private transient List<NestedField> fields = null;
 
     private MapType(NestedField keyField, NestedField valueField) {
+      Preconditions.checkArgument(
```

Comment:
what about the keyField type? or is that validation already in place?

Reply:
Yes, the when initializing a `MapType` with an `UnknownType` as the key, it will throw right away because the key has to be required.